pkg/sync/sync_context.go

```diff
@@ -289,6 +299,19 @@ func (sc *syncContext) Sync() {
 		}
 	}
 
+	//This is to support namespace creation created by pre-sync task.
```

Comment:
I think it is better to load live namespace resource at the same [place](https://github.com/argoproj/gitops-engine/pull/94/files#diff-f6928024340969ce58a20e045c7dc520R554) where we inject sync task. This way code related to namespace auto-creation won't be spread.

Reply:
Moved code as close as it can be now. :)